File: unit4/DSA/backtracking/subset.js
function odd_subset(n,arr,k,i,res) {
  if (i == n) {
    return 
  }
  res.push(arr[i])
  console.log(res)
  odd_subset(n,arr,k,i+1,res)
  res.pop()
  odd_subset(n,arr,k,i+1,res)

}
let n=3
let arr=[1,2,3]
let k=2
let res =[]
let ans = odd_subset(n,arr,k,0,res)

File: unit4/DSA/backtracking/practce/subset.py
def subset(n,s,i,curr):  #0  1  
    if i==n:
        return
    curr.append(s[i])
    print(curr)
    subset(n,s,i+1,curr)
    curr.pop()
    subset(n,s,i+1,curr)

n=3
s="abc"
ans = subset(n,s,0,[])
print(ans)

